homework/shared_ptr/test.cpp

```diff
@@ -97,24 +93,24 @@ TEST_F(SharedPointerTestSuite, dereferencingAfterGet)
 
 TEST_F(SharedPointerTestSuite, asgnOperator)
 {
-    my_shared_ptr<SomeUsefulClass> copy_ptr = m_sut;
+    auto copy_ptr = my_shared_ptr<SomeUsefulClass>(new SomeUsefulClass{});
+    copy_ptr = m_sut;  // this calls copy assignment operator
     EXPECT_EQ(copy_ptr.use_count(), 2);
 }
 
-TEST_F(SharedPointerTestSuite, CopyConstructor)
+TEST_F(SharedPointerTestSuite, CopyConstructorAndBoolConversion)
 {
     my_shared_ptr<int> ptr = my_shared_ptr<int>(new int(initial_value));
     ASSERT_TRUE(ptr);
     my_shared_ptr<int> copy_ptr = my_shared_ptr<int>(ptr);
     ASSERT_TRUE(copy_ptr);
```

Comment:
add a case when conversion gives false
```suggestion
    my_shared_ptr<int> copy_ptr = my_shared_ptr<int>(ptr);
    ASSERT_TRUE(copy_ptr);
    copy_ptr.reset();
    ASSERT_FALSE(copy_ptr);
```